torch/fx/experimental/symbolic_shapes.py

```diff
@@ -407,10 +428,10 @@ def create_symbolic_sizes_strides(self, ex: torch.Tensor):
                 )[0]
                 stride[i] = self.create_symbol(val)
         assert all(x is not None for x in stride)
-        return [self.create_symintnode(i) for i in size], [self.create_symintnode(i) for i in stride]  # type: ignore[arg-type]
+        return [self.create_symintnode(x, id(ex), "size", idx) for idx, x in enumerate(size)], [self.create_symintnode(x, id(ex), "stride", idx) for idx, x in enumerate(stride)]  # type: ignore[arg-type] # noqa: B950
```

Comment:
The use of id() here is puzzling to me; naively, what I would have expected is that Dynamo tells you the index of every argument into the graph, and that's what you correlate guards back to, so that when you are testing entry into dynamo, you just have your array of tensors and you index into it to find the new ones.

The use of id feels wrong, because the next time you hit a subgraph, I wouldn't necessarily expect the id of all input tensors to be the same, but I still want to test the same guards on the "second" tensor argument, e.g.

Reply:
> The use of id feels wrong, because the next time you hit a subgraph, I wouldn't necessarily expect the id of all input tensors to be the same, but I still want to test the same guards on the "second" tensor argument, e.g.

Yeah, that is true. The problem is when the compared for elements don't come from input tensors, but weights in the graph. Let me see what I can do with that...

Reply:
> The use of id() here is puzzling to me; naively, what I would have expected is that Dynamo tells you the index of every argument into the graph, and that's what you correlate guards back to, so that when you are testing entry into dynamo, you just have your array of tensors and you index into it to find the new ones.
> 
> The use of id feels wrong, because the next time you hit a subgraph, I wouldn't necessarily expect the id of all input tensors to be the same, but I still want to test the same guards on the "second" tensor argument, e.g.

We don't use id in the guards. We use id to map to tensor name, so it does what you said. However, its not only inputs, its model states too, atm.